File: 351002/Lashchenka/publisher/src/main/java/by/bsuir/distcomp/dto/request/TweetRequestTo.java
package by.bsuir.distcomp.dto.request;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

public class TweetRequestTo {
    private Long id;

    @NotNull(message = "Editor ID must not be null")
    private Long editorId;

    @NotBlank(message = "Title must not be blank")
    @Size(min = 2, max = 64, message = "Title must be between 2 and 64 characters")
    private String title;

    @NotBlank(message = "Content must not be blank")
    @Size(min = 4, max = 2048, message = "Content must be between 4 and 2048 characters")
    private String content;

    public TweetRequestTo() {}

    public TweetRequestTo(Long id, Long editorId, String title, String content) {
        this.id = id;
        this.editorId = editorId;
        this.title = title;
        this.content = content;
    }

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Long getEditorId() { return editorId; }
    public void setEditorId(Long editorId) { this.editorId = editorId; }
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    public String getContent() { return content; }
    public void setContent(String content) { this.content = content; }
}

File: 351002/Khmylko/rest-api/publisher/src/DTO/Request/TweetRequestTo.php
<?php
namespace App\DTO\Request;

class TweetRequestTo {
    private int $editorId;
    private string $title;
    private string $content;

    public function __construct(array $data) {
        $this->editorId = $data['editorId'] ?? 0;
        $this->title = $data['title'] ?? '';
        $this->content = $data['content'] ?? '';
    }

    public function getEditorId(): int { return $this->editorId; }
    public function getTitle(): string { return $this->title; }
    public function getContent(): string { return $this->content; }
}